delta_filter.cpp

```diff
@@ -120,48 +134,116 @@ DeltaFilter::DeltaData::~DeltaData()
 }
 
 /**
- * Evaluate a reading to determine if it needs to be sent
- * The conditions that cause it to be sent are:
+ * Check whether tolerance is exceeded given old and new DatapointValue objects
  *
- *	The minimum rate, if set requires a value to be sent
+ * @param oValue		old DatapointValue
+ * @param nValue		new DatapointValue
+ * @param toleranceMeasure	measure of tolerance: percentage or absolute value
+ * @param tolerance		tolerance percentage or absolute value
+ * @param change		returns absolute percentage or absolute change in Datapoint values
+ * @return bool         whether tolerance was exceeded
+ */
+bool checkToleranceExceeded(const string &dpName, const DatapointValue& oValue, const DatapointValue& nValue, 
```

Comment:
Can't help thinking a bit of judicious use of classes for the tolerance configuration could make this a whole lot more manageable.  But I suspect this is begin done under time pressure, so probably not the time to think about it.

Reply:
Yes, I agree.